pkg/resources/validation/project/project_test.go

```diff
@@ -0,0 +1,174 @@
+				},
+			},
+			oldProject: apisv3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "anyproject",
+					Labels: map[string]string{
+						validation.EnforceLabel: "baseline",
+					},
+				},
+			},
+			wantErr: false,
+			allowed: true,
+		},
+		{
+			name:        "Update PSA for NON admin user-should not be allowed",
+			userName:    failSarUser,
+			clusterName: "validcluster",
+			project: apisv3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "anyproject",
+					Labels: map[string]string{
+						validation.EnforceLabel: "baseline",
+					},
+				},
+			},
+			oldProject: apisv3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "anyproject",
+					Labels: map[string]string{},
+				},
+			},
+			wantErr: false,
+			allowed: false,
+		},
+	}
+
+	k8Fake := &k8testing.Fake{}
+	fakeSAR := &k8fake.FakeSubjectAccessReviews{Fake: &k8fake.FakeAuthorizationV1{Fake: k8Fake}}
+	validator := project.NewValidator(fakeSAR)
+
+	k8Fake.AddReactor("create", "subjectaccessreviews", func(action k8testing.Action) (handled bool, ret runtime.Object, err error) {
+		createAction := action.(k8testing.CreateActionImpl)
+		review := createAction.GetObject().(*authrizationv1.SubjectAccessReview)
+		spec := review.Spec
+
+		// simulate SAR not allowed
+		if spec.User == failSarUser {
+			review.Status.Allowed = false
+			review.Status.Reason = fmt.Sprintf("%s %s", "Can not update project PSA for:", spec.User)
+			return true, review, nil
+		}
+		review.Status.Allowed = true
+		return true, review, nil
+	})
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			request := createProjectRequest(t, &tt)
+			resp, err := validator.Admit(request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Admit() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !resp.Allowed == tt.allowed {
+				t.Errorf("Admit() got = %v, want %v", resp, tt.allowed)
+			}
+		})
+	}
+}
+
+func createProjectRequest(t *testing.T, s *test) *admission.Request {
+	t.Helper()
+	gvk := metav1.GroupVersionKind{Group: "management.cattle.io", Version: "v3", Kind: "Project"}
+	gvr := metav1.GroupVersionResource{Group: "management.cattle.io", Version: "v3", Resource: "project"}
+	oldobj, _ := json.Marshal(s.oldProject)
+	req := &admission.Request{
+		AdmissionRequest: admissionv1.AdmissionRequest{
+			UID:             "",
+			Kind:            gvk,
+			Resource:        gvr,
+			RequestKind:     &gvk,
+			RequestResource: &gvr,
+			Name:            s.project.Name,
+			Namespace:       s.clusterName,
+			Operation:       admissionv1.Update,
+			UserInfo:        authenticationv1.UserInfo{Username: s.userName, UID: ""},
+			Object:          runtime.RawExtension{},
+			OldObject: runtime.RawExtension{
+				Raw: oldobj,
+			},
+			Options: runtime.RawExtension{},
+		},
+		Context: context.Background(),
+	}
+	var err error
+	req.Object.Raw, err = json.Marshal(s.project)
+	assert.NoError(t, err, "Failed to marshal Project while creating request")
```

Comment:
`require` will stop the test `assert` will continue execution.
```suggestion
	require.NoError(t, err, "Failed to marshal Project while creating request")
```